frontend/src/Teacher/AllTasks.vue

```diff
@@ -0,0 +1,202 @@
+<script setup lang="ts">
+import type { Config, ConfigColumns, Api as DataTableObject } from 'datatables.net';
+import DataTablesCore from 'datatables.net-bs5';
+import DataTable from 'datatables.net-vue3';
+import { ref, watchEffect } from 'vue';
+import Loader from '../components/Loader.vue';
+import { getDate, getFromAPI } from '../utilities';
+
+DataTable.use(DataTablesCore);
+
+type Task = {
+  id: string;
+  title: string;
+  subject: string;
+  date: Date;
+};
+
+type RawTask = Omit<Task, 'date'> & {
+  date: string;
+};
+
+type SortValue = 'asc' | 'desc';
+
+/**
+ * Get tasks from API
+ * @param subject Subject abbreviation
+ * @param count Count of records
+ * @param start Offset of records
+ * @param sort Type of sorting
+ * @param search Search query
+ * @returns Tuple of [total count, tasks]
+ */
+const getTasks = async (
+  subject: string,
+  count: number,
+  start = 0,
+  sort: SortValue = 'desc',
+  search: string = ''
+): Promise<[number, Task[]]> => {
+  const params = new URLSearchParams();
+
+  if (subject !== 'all') {
+    params.append('subject', subject);
+  }
+
+  params.append('count', count.toString());
+  params.append('start', start.toString());
+  params.append('sort', sort);
+  params.append('search', search);
+
+  const data = await getFromAPI<{
+    tasks: RawTask[];
+    count: number;
+  }>(`/api/tasks/all?${params.toString()}`);
+
+  if (data) {
+    return [
+      data.count,
+      data.tasks.map((task) => {
+        return {
+          ...task,
+          //parse date, and since django returns the correct format we can just pass it to the Date constructor
+          date: new Date(task.date)
+        } satisfies Task;
+      })
+    ];
+  }
+
+  return [0, []];
+};
+
+type Subject = {
+  abbr: string;
+  name: string;
+};
+
+const subjects = ref(Array<Subject>());
+
+const loaded = ref(false);
+const getSubjects = async () => {
+  const data = await getFromAPI<{
+    subjects: Subject[];
+  }>('/api/subjects/all');
+
+  if (data) {
+    subjects.value = data.subjects;
+    loaded.value = true;
+  }
+};
+
+getSubjects();
```

Comment:
This doesn't need to happen in this PR, but just an idea: we essentially need to load the subjects before displaying this component. We could use `await getSubjects()` + https://vuejs.org/guide/built-ins/suspense to avoid dealing with missing subject data, to make the code a bit easier.

Reply:
So I don't need to show loading at all? This will make getting datatable instance easier.

Reply:
Yes, the component would not even start to render until the async data get loaded, which simplifies the code, because it avoid having to deal with `null`/missing values that are loaded asynchronously and that are required for the component to work.

The issue then of course becomes where does the loader component get moved to. With `Suspense`, it gets moved to the parent component. I'm not sure if it is feasible to use Suspense with the web component method that we use right now, though.